test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -540,7 +540,16 @@ private void wipeCluster() throws Exception {
                 // All other repo types we really don't have a chance of being able to iterate properly, sadly.
                 Request listRequest = new Request("GET", "/_snapshot/" + repoName + "/_all");
                 listRequest.addParameter("ignore_unavailable", "true");
-                List<?> snapshots = (List<?>) entityAsMap(adminClient.performRequest(listRequest)).get("snapshots");
+
+                Map<?, ?> response = entityAsMap(adminClient.performRequest(listRequest));
+                Map<?, ?> oneRepoResponse;
+                if (response.containsKey("responses")) {
```

Comment:
Shouldn't we actually check all snapshots from all repositories here now instead of simply taking the first repo and ignore the rest arbitrarily?

Reply:
There is an outer loop that iterates over all the repositories. Probably I can replace it with a single API call to get all snapshots from all the repositories. But this line worries me
```
if (false == preserveSnapshotsUponCompletion() && repoType.equals("fs")) {
                // All other repo types we really don't have a chance of being able to iterate properly, sadly.
```
So it seems that we want to wipe snapshots from fs repositories only and _all won't do the trick.

Reply:
Ah I see now, don't worry about it then imo, it would just be an optimization :) Didn't notice/remember the outer loop :)